packages/odata-common/src/batch-response.ts

```diff
@@ -1,51 +1,59 @@
-import { DefaultDeSerializers, DeSerializers } from './de-serializers';
+import { DeSerializers } from './de-serializers';
 import { EntityApi, EntityBase } from './entity-base';
 /**
  * @internal
  */
-export type BatchResponse<
-  DeSerializersT extends DeSerializers = DefaultDeSerializers
-> =
+export type BatchResponse<DeSerializersT extends DeSerializers> =
+  | ErrorResponse
   | ReadResponse<DeSerializersT>
-  | WriteResponses<DeSerializersT>
-  | ErrorResponse;
+  | WriteResponses<DeSerializersT>;
+
+interface SuccessTypeGuard<DeSerializersT extends DeSerializers> {
+  isSuccess: () => this is
+    | ReadResponse<DeSerializersT>
+    | WriteResponses<DeSerializersT>;
+}
+
+interface ReadRespneTypeGuard<DeSerializersT extends DeSerializers> {
```

Comment:
I am missing a `WriteResponsesTypeGuard` a bit.
I actually think it would be nice to have one type guard for all, e.g. `BatchResponseTypeGuard` that has all four functions (`isSuccess`, `isError`, `isReadReponse`, `isWriteResponses`), so that as a user I just need one function call to get what I need. Currently I always have to call `isSuccess()` first and if I am looking for a write response I have to know that `!isReadResponse()` will give me that.